File: UDP--/add.c
#include <stdio.h>


int main(){
    float i=0.1;
    int c;
    for(c=0;c<10000000;c++){
        i+=0.2;
    }
    printf("result: %f\n", i);
    return 0;
}

// result: 2175874.250000 差的真多

File: UDP--/add.py
#!/usr/bin/env python3
# coding=utf-8
# date 2022-12-14 10:42:33
# author calllivecn <c-all@qq.com>

import time

import numba

def f1(n):
    i=0.1
    for _ in range(n):
        i+=0.2
    print("result: %f", i)

@numba.jit
def f2(n):
    i=0.1
    for _ in range(n):
        i+=0.2
    print("result: %f", i)

count = 1000000000

t1 = time.time()
f1(count)
t2 = time.time()
print("耗时：", t2 -t1, "/秒")


t1 = time.time()
f2(count)
t2 = time.time()
print("耗时：", t2 -t1, "/秒")

